release-tools/stochss_ami_manager.py

```diff
@@ -0,0 +1,547 @@
+#!/usr/bin/env python
+
+__author__ = 'Dibyendu Nath'
+__email__ = 'dnath@cs.ucsb.edu'
+
+import sys
+import os
+import boto.ec2
+import time
+import uuid
+import json
+import threading
+import subprocess
+import tempfile
+import traceback
+import pprint
+import argparse
+import glob
+
+
+def get_remote_command(user, ip, key_file, command):
+    return 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i {0} {1}@{2} "{3}"'.format(key_file, user,
+                                                                                                         ip, command)
+
+
+class ShellCommand(object):
+    def __init__(self, cmd, stdin=sys.stdin, stdout=sys.stdout, stderr=sys.stderr, verbose=False):
+        self.cmd = cmd
+        self.stdin = stdin
+        self.stdout = stdout
+        self.stderr = stderr
+        self.process = None
+        self.verbose = verbose
+
+    def run(self, timeout=None, silent=False):
```

Comment:
This function should check the return value of the executed process.  If it is not 'success', it should raise a custom Exception instead of returning.